core/src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -469,6 +490,11 @@ Path newDistroFile(Environment env) throws IOException {
             return Files.createTempFile(env.tmpFile(), name, ".zip");
         }
 
+        Tuple<URL, Path> newChecksumUrlAndFile(Environment env, URL originalUrl, String suffix) throws IOException {
+            URL newUrl = new URL(originalUrl.toString() + "." + suffix);
```

Comment:
+1 to get rid of URL and use URI instead, as a follow-up. Opened https://github.com/elastic/elasticsearch/issues/12896 for this.